SwiftAdvanceExercise2.playground/Pages/Extensions_6.1.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+
+extension String {
+    //Extension exercise 1
+    mutating func insertCharAt(char: String, at: Int) {
+        self.insert(contentsOf: char, at: self.index(self.startIndex, offsetBy: at) )
+    }
+    
+    //Extension ecercise 2 & 3
+    mutating func replaceCharAt(old: String, new: String) {
+        self = self.replacingOccurrences(of: old, with: new)
+    }
+    
+    //Extension ecercise 4
+    func counWords() -> Int {
+        let word = self.components(separatedBy: " ")
+        return word.count
```

Comment:
we can create a computed property
var wordCount: String {
    self.components(separatedBy: " ")
}